drivers/haptics/drv2605.c

```diff
@@ -498,7 +498,10 @@ static int drv2605_reset(const struct device *dev)
 	k_msleep(100);
 
 	while (retries > 0) {
-		i2c_reg_read_byte_dt(&config->i2c, DRV2605_REG_MODE, &value);
+		ret = i2c_reg_read_byte_dt(&config->i2c, DRV2605_REG_MODE, &value);
```

Comment:
IIRC this change will break the driver since what we are doing here is polling after a hardware reset until the part is awake and the control port is accessible.

There is a bug here however. `value` is uninitialized until there is a successful `i2c_reg_read_byte_dt`  call, so I think what we should do here is check the return from that call and only check `DRV2605_DEV_RESET` upon a successful transaction or retry otherwise

Reply:
so the solution would be:

```
retries--;
ret = i2c_reg_read_byte_dt(&config->i2c, DRV2605_REG_MODE, &value);
if(ret<0){
    continue;
}
```

Reply:
@jilaypandya, I like that solution.